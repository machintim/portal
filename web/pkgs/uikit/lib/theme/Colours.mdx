import { Meta, Title, Subtitle, ColorPalette, ColorItem } from '@storybook/addon-docs/blocks';
import theme from './defaultTheme';

<Meta title="theme/Colours" />
<Title>Colours explorer</Title>
<Subtitle>Colours approved for use in the UCL Design System</Subtitle> 

More information on the use of colour in the UCL Design System <a href="https://www.figma.com/design/8Sm5PxWOWJYpYXAzhRzUzt/UCL-Design-System-UI-Kit?node-id=1-287" target="_blank">can be found on Figma</a>. 

The colours populating this page are imported from the <a href="https://github.com/ucl-isd/uikit-react/blob/main/lib/theme/defaultTheme.ts#L5" target="_blank">theme file</a>. 

Below are all colours per category, one category per line: 

<ColorPalette>
{
  Object.entries(theme.color).map(([key, value]) => {
    return (
      <ColorItem key={key} title={key.charAt(0).toUpperCase() + key.slice(1)} colors={value} />
    );
  })
}
</ColorPalette>

Below is a breakdown of each colour category. These are the same colours as above, but grouped by category, one colour per line: 

{Object.entries(theme.color).map(([key, value]) => {
  return (
    <div key={key}>
      <h2>{key.charAt(0).toUpperCase() + key.slice(1)}</h2>

      <ColorPalette>

      {Object.entries(value).map(([colorKey, colorValue]) => {
        return (
          <ColorItem key={colorKey} title={colorKey} colors={{ [colorKey]: colorValue }} />
        );
      })}

      {console.log(key, value)}

      </ColorPalette>
    </div>
  );
})}
