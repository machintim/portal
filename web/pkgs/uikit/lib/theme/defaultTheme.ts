// UCL Theme

// base colours

const purple200 = '#2C0442'; // UCL dark purple
const purple50 = '#500778'; // UCL vibrant purple
const purple40 = '#6345A5';
const purple20 = '#BBAACC';
// TODO: where is UCL muted purple = #C6B0BC ?
const green100 = '#006B0F';
const green70 = '#008613';
const green40 = '#80C289';
const green10 = '#E6F3E8';
const green200 = '#113B3A'; // UCL dark green
const green50 = '#52C152'; // UCL vibrant green
const green20 = '#C9D1A8'; // UCL muted green

const blue100 = '#004089';
const blue80 = '#004EA8';
const blue70 = '#0D68CF';
const blue40 = '#7FB7F7';
const blue10 = '#E5F0FD';
const blue5 = '#F2F8FE';
const blue200 = '#002248'; // UCL dark blue
const blue50 = '#34C6C6'; // UCL vibrant blue
const blue20 = '#B6DCE5'; // UCL muted blue

const yellow200 = '#504D48'; // UCL dark yellow
const yellow50 = '#FFCA36'; // UCL vibrant yellow
const yellow20 = '#DAD6CA'; // UCL muted yellow

const orange100 = '#CC7100';
const orange70 = '#FF8D00';
const orange40 = '#FFC680';
const orange10 = '#FFF4E6';

const red100 = '#811712';
const red70 = '#CA0007';
const red50 = '#E48083';
const red40 = '#E48083';
const red10 = '#FAE6E7';

const pink200 = '#4B0A32'; // UCL dark pink
const pink50 = '#AC145A'; // UCL vibrant pink
const pink20 = '#DEB8C3'; // UCL vibrant pink

const black100 = '#000000';
const black90 = '#1A1A1A';
const black80 = '#333333';
const black60 = '#666666';
const black40 = '#999999';
const black20 = '#CCCCCC';
const black15 = '#D9D9D9';
const black10 = '#E6E6E6';
const black5 = '#F2F2F2';

const white = '#ffffff';

const baseColour = {
  purple200,
  purple50,
  purple40,
  purple20,

  green100,
  green70,
  green40,
  green10,
  green200,
  green50,
  green20,

  blue100,
  blue80,
  blue70,
  blue40,
  blue10,
  blue5,
  blue200,
  blue50,
  blue20,

  yellow200,
  yellow50,
  yellow20,

  orange100,
  orange70,
  orange40,
  orange10,
  // missing orange50?
  red100,
  red70,
  red50,
  red40,
  red10,

  pink200,
  pink50,
  pink20,

  black100,
  black90,
  black80,
  black60,
  black40,
  black20,
  black15,
  black10,
  black5,

  white,
};

const textColours = {
  primary: baseColour.black90, // Primary text, such as body copy, headers, displays, labels.
  secondary: baseColour.black60, // Secondary text, such as stand-first, paragraph summary, breadcrumbs.
  muted: baseColour.black40, // Tertiary text, such as input fields placeholder.
  inverted: baseColour.white, // Text on bold background, for example primary buttons label.

  interactive: baseColour.blue70, // Link, secondary button.
  default: baseColour.black60, // Status default for component label: tabs, toolbar.
  disabledOnBg: baseColour.black40, // Status disabled on background.
  disabled: baseColour.black20, // Status disabled.

  active: baseColour.blue70, // Status active for components' label: tabs, toolbar.
  hover: baseColour.blue100, // Status hover for links and components' label.
  visited: baseColour.purple40, // Status visited for links.

  error: baseColour.red70, // Label for buttons or link with destructive purpose.
  errorHover: baseColour.red100, // Error hover.
  errorDisabled: baseColour.red50, // Error disabled.
};

const interactionColours = {
  blue100: baseColour.blue100,
  blue80: baseColour.blue80,
  blue70: baseColour.blue70,
  blue40: baseColour.blue40,
  blue10: baseColour.blue10,
  blue5: baseColour.blue5,
};

const displayColours = {
  purpleDark: baseColour.purple200,
  purpleVibrant: baseColour.purple50,
  purpleMuted: baseColour.purple20,

  greenDark: baseColour.green200,
  greenVibrant: baseColour.green50,
  greenMuted: baseColour.green20,

  yellowDark: baseColour.yellow200,
  yellowVibrant: baseColour.yellow50,
  yellowMuted: baseColour.yellow20,

  pinkDark: baseColour.pink200,
  pinkVibrant: baseColour.pink50,
  pinkMuted: baseColour.pink20,

  blueDark: baseColour.blue200,
  blueVibrant: baseColour.blue50,
  blueMuted: baseColour.blue20,
};

const systemColours = {
  red100: baseColour.red100,
  red70: baseColour.red70,
  red40: baseColour.red40,
  red10: baseColour.red10,

  orange100: baseColour.orange100,
  orange70: baseColour.orange70,
  orange40: baseColour.orange40,
  orange10: baseColour.orange10,

  green100: baseColour.green100,
  green70: baseColour.green70,
  green40: baseColour.green40,
  green10: baseColour.green10,

  blue100: baseColour.blue100,
  blue80: baseColour.blue80,
  blue70: baseColour.blue70,
  blue40: baseColour.blue40,
  blue10: baseColour.blue10,
  blue5: baseColour.blue5,
};

const linkColours = {
  default: baseColour.blue70,
  hover: baseColour.blue100,
  visited: baseColour.purple40,
  disabled: baseColour.black20,
};

const blackAndWhite = {
  black: baseColour.black100,
  white: baseColour.white,
};

const neutralColours = {
  black: blackAndWhite.black,
  grey90: baseColour.black90,
  grey80: baseColour.black80,
  grey60: baseColour.black60,
  grey40: baseColour.black40,
  grey20: baseColour.black20,
  grey15: baseColour.black15,
  grey10: baseColour.black10,
  grey5: baseColour.black5,
  white: blackAndWhite.white,
};

const overlayColours = {
  blanket: 'rgba(0, 0, 0, 0.5)',
};

const color = {
  text: textColours,
  interaction: interactionColours,
  display: displayColours,
  system: systemColours,
  link: linkColours,
  neutral: neutralColours,
  overlay: overlayColours,
};

// These colour definitions will need to be refactored to match the design system.

const baseSizes = {
  s1: '1px',
  s2: '2px',
  s4: '4px',
  s6: '6px',
  s8: '8px',
  s12: '12px',
  s16: '16px',
  s20: '20px',
  s24: '24px',
  s28: '28px',
  s32: '32px',
  s36: '36px',
  s40: '40px',
  s44: '44px',
  s48: '48px',
  s52: '52px',
  s56: '56px',
  s60: '60px',
  s64: '64px',
  s80: '80px',
};

const margin = {
  m2: baseSizes.s2,
  m4: baseSizes.s4,
  m6: baseSizes.s6,
  m8: baseSizes.s8,
  m12: baseSizes.s12,
  m16: baseSizes.s16,
  m20: baseSizes.s20,
  m24: baseSizes.s24,
  m32: baseSizes.s32,
  m40: baseSizes.s40,
  m48: baseSizes.s48,
  m56: baseSizes.s56,
  m64: baseSizes.s64,
};

const padding = {
  p2: baseSizes.s2,
  p4: baseSizes.s4,
  p6: baseSizes.s6,
  p8: baseSizes.s8,
  p12: baseSizes.s12,
  p16: baseSizes.s16,
  p20: baseSizes.s20,
  p24: baseSizes.s24,
  p32: baseSizes.s32,
  p40: baseSizes.s40,
  p48: baseSizes.s48,
  p56: baseSizes.s56,
  p64: baseSizes.s64,
};

const breakpoints = {
  mobile: 320,
  tablet: 672,
  desktop: 1056,
  desktopLarge: 1536,
};

const createMediaQuery = (breakpoint: number) =>
  // Breakpoints apply at the specified width and above
  `@media (min-width: ${breakpoint}px)`;

const custom = (breakpoint: number) =>
  // Fallback for edge cases
  createMediaQuery(breakpoint);

const mq = {
  mobile: createMediaQuery(breakpoints.mobile),
  tablet: createMediaQuery(breakpoints.tablet),
  desktop: createMediaQuery(breakpoints.desktop),
  desktopLarge: createMediaQuery(breakpoints.desktopLarge),
  custom,
};

const theme = {
  color,

  thickness: {
    t1: baseSizes.s1,
    t2: baseSizes.s2,
  },

  width: {
    h16: baseSizes.s16,
    h20: baseSizes.s20,
    h24: baseSizes.s24,
  },

  height: {
    h16: baseSizes.s16,
    h20: baseSizes.s20,
    h24: baseSizes.s24,
    h32: baseSizes.s32,
    h36: baseSizes.s36,
    h40: baseSizes.s40,
    h80: baseSizes.s80,
  },

  margin,

  padding,

  radius: {
    r2: baseSizes.s2,
    r4: baseSizes.s4,
    r8: baseSizes.s8,
    r12: baseSizes.s12,
    r16: baseSizes.s16,
  },

  border: {
    b1: baseSizes.s1,
    b2: baseSizes.s2,
  },

  outline: {
    o2: baseSizes.s2,
    o4: baseSizes.s4,
  },

  font: {
    family: {
      primary: 'Helvetica, "Helvetica Neue", Arial, sans-serif',
      secondary:
        'Garamond, Baskerville, "Baskerville Old Face", "Hoefler Text", "Times New Roman", serif',
      fixed: 'Courier, fixed',
    },
    size: {
      f12: '12px',
      f14: '14px', // Font should be at least 14px in most cases
      f16: '16px', // Prefer 16px to 14px in most cases
      f18: '18px',
      f20: '20px',
      f22: '22px',
      f24: '24px',
      f26: '26px',
      f28: '28px',
      f30: '30px',
      f32: '32px',
      f36: '36px',
      f40: '40px',
    },
    weight: {
      light: 300,
      regular: 400,
      medium: 500,
      semiBold: 600,
      bold: 700,
      extraBold: 800,
    },
    lineHeight: {
      h100: '100%',
      h110: '110%',
      h120: '120%',
      h130: '130%',
      h140: '140%',
      h150: '150%',
    },
  },

  boxShadow: {
    // offset-x | offset-y | blur-radius | spread-radius | color
    y1: '0 1px 4px 1px rgba(0, 0, 0, 0.1)',
    y2: '0 2px 4px 1px rgba(0, 0, 0, 0.1)',
    x2y4: '2px 4px 8px 4px rgba(0, 0, 0, 0.1)',
    focus: `0px 0px 0px 2px ${color.neutral.white}, 0px 0px 0px 4px ${color.neutral.black}, 0px 0px 0px 6px ${color.system.orange70}`,
    active: `0px 6px 0px 0px ${color.interaction.blue80} inset`,
    destructiveActive: `0px 6px 0px 0px ${color.system.red100} inset`,
  },

  breakpoints,
  mq,

  // For ease of use, margin and padding are added at the top level
  // Usage: margin-top: ${theme.m16};
  ...margin,
  ...padding,
};

export type ThemeType = typeof theme;
export type DisplayColoursType = typeof displayColours;
export type BlackAndWhiteColoursType = typeof blackAndWhite;

export default theme;
