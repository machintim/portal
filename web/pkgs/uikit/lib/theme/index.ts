export { default as theme } from './defaultTheme';
export type {
  ThemeType,
  DisplayColoursType,
  BlackAndWhiteColoursType,
} from './defaultTheme';
export {
  default as useTheme,
  ThemeContextProvider,
} from './useTheme';
