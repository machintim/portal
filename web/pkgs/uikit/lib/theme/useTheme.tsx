// We will probably move away from context/provider to store/hook for accessing theme.

import {
  createContext,
  useState,
  useContext,
  ReactNode,
} from 'react';

import theme, { ThemeType } from './defaultTheme';

const useThemeState = () => useState(theme as ThemeType);

const ThemeContext = createContext<ReturnType<
  typeof useThemeState
> | null>(null);

export const ThemeContextProvider: React.FC<{
  children: ReactNode;
}> = ({ children }) => (
  <ThemeContext.Provider value={useThemeState()}>
    {children}
  </ThemeContext.Provider>
);

const useTheme = () => {
  const value = useContext(ThemeContext);
  if (value === null) {
    throw new Error('Please add ThemeContextProvider');
  }
  return value;
};

export default useTheme;
