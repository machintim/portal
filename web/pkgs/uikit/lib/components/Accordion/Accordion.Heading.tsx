import React, { HTMLAttributes, JSX } from 'react';
import { css, cx } from '@emotion/css';
import useTheme from '../../theme/useTheme';
import Icon from '../Icon/Icon';
import { useAccordionContext } from './Accordion';

export const NAME = 'ucl-accordion__heading';

interface AccordionHeadingProps
  extends HTMLAttributes<HTMLElement> {
  as?: keyof JSX.IntrinsicElements;
  testId?: string;
}

const AccordionHeading: React.FC<AccordionHeadingProps> = ({
  children,
  as: Component = 'div',
  testId = NAME,
  className,
}) => {
  const [theme] = useTheme();
  const { isOpen, toggleAccordion, disabled, size } =
    useAccordionContext();

  const headingStyle = cx(
    NAME,
    css`
      display: flex;
      justify-content: space-between;
      align-items: center;
      padding: ${theme.padding.p16};
      background-color: ${theme.color.neutral.white};
      cursor: ${disabled ? 'not-allowed' : 'pointer'};
      font-weight: ${theme.font.weight.bold};
      transition:
        background-color 0.3s ease,
        box-shadow 0.3s ease;

      &:hover {
        background-color: ${!disabled
          ? theme.color.neutral.grey5
          : theme.color.text.disabled};
      }

      &:focus-visible {
        outline: none;
        box-shadow: ${theme.boxShadow.focus};
      }
    `,
    className
  );

  return (
    <Component
      className={headingStyle}
      data-testid={testId}
      onClick={!disabled ? toggleAccordion : undefined}
      role='button'
      tabIndex={0}
      aria-expanded={isOpen}
    >
      {children}
      <Icon.ChevronDown
        size={size === 'medium' ? 32 : 24}
        className={css`
          transform: rotate(${isOpen ? '180deg' : '0deg'});
          transition: transform 0.15s ease;
          color: ${theme.color.interaction.blue70};
        `}
      />
    </Component>
  );
};

export default AccordionHeading;
