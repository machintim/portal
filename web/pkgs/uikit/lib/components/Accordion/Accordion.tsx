import React, {
  createContext,
  useContext,
  useState,
  HTMLAttributes,
} from 'react';
import AccordionHeading from './Accordion.Heading';
import AccordionPanel from './Accordion.Panel';
import { css, cx } from '@emotion/css';
import useTheme from '../../theme/useTheme';

export const NAME = 'ucl-accordion';

interface AccordionContextProps {
  isOpen: boolean;
  toggleAccordion: () => void;
  disabled: boolean;
  size: 'small' | 'medium';
}

const AccordionContext = createContext<
  AccordionContextProps | undefined
>(undefined);

export const useAccordionContext = () => {
  const context = useContext(AccordionContext);
  if (!context) {
    throw new Error(
      'useAccordionContext must be used within an AccordionProvider'
    );
  }
  return context;
};

export interface AccordionProps
  extends HTMLAttributes<HTMLDivElement> {
  testId?: string;
  size?: 'small' | 'medium';
  disabled?: boolean;
  isOpen?: boolean;
}

const Accordion: React.FC<AccordionProps> & {
  Heading: typeof AccordionHeading;
  Panel: typeof AccordionPanel;
} = ({
  size = 'medium',
  disabled = false,
  isOpen = false,
  children,
  className,
  testId = NAME,
}) => {
  const [open, setOpen] = useState(isOpen);
  const [theme] = useTheme();

  const toggleAccordion = () => {
    if (!disabled) {
      setOpen((prev) => !prev);
    }
  };

  const accordionStyle = cx(
    NAME,
    css`
      max-width: 600px;
      border: 1px solid ${theme.color.neutral.grey10};
      background-color: ${theme.color.neutral.white};
      font-family: ${theme.font.family.primary};
      font-size: ${theme.font.size.f16};
    `,
    size === 'small' &&
      css`
        font-size: ${theme.font.size.f16};
        line-height: ${theme.height.h24};
      `,
    size === 'medium' &&
      css`
        font-size: ${theme.font.size.f20};
      `,
    disabled &&
      css`
        pointer-events: none;
        background-color: ${theme.color.link.disabled};
        color: ${theme.color.text.muted};
        // border-color: ${theme.color.text.muted};
      `,
    className
  );

  return (
    <AccordionContext.Provider
      value={{
        isOpen: open,
        toggleAccordion,
        disabled,
        size,
      }}
    >
      <div
        className={accordionStyle}
        aria-disabled={disabled}
        data-testid={testId}
      >
        {children}
      </div>
    </AccordionContext.Provider>
  );
};

Accordion.Heading = AccordionHeading;
Accordion.Panel = AccordionPanel;

export default Accordion;
