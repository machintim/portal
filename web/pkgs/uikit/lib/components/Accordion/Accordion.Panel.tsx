import React, { HTMLAttributes } from 'react';
import { css, cx } from '@emotion/css';
import useTheme from '../../theme/useTheme';
import { useAccordionContext } from './Accordion';

export const NAME = 'ucl-accordion__panel';

interface AccordionPanelProps
  extends HTMLAttributes<HTMLDivElement> {
  testId?: string;
}

const AccordionPanel: React.FC<AccordionPanelProps> = ({
  children,
  className,
  testId = NAME,
}) => {
  const [theme] = useTheme();
  const { isOpen } = useAccordionContext();

  if (!isOpen) return null;

  const panelStyle = cx(
    NAME,

    css`
      padding: ${theme.padding.p16};
      font-family: ${theme.font.family.primary};
      font-weight: ${theme.font.weight.regular};
      background-color: ${theme.color.neutral.white};
      font-size: ${theme.font.size.f16};
      color: ${theme.color.text.primary};
    `,
    className
  );

  return (
    <div
      className={panelStyle}
      data-testid={testId}
    >
      {children}
    </div>
  );
};

export default AccordionPanel;
