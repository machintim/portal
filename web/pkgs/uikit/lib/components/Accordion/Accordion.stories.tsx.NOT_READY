import type { Meta, StoryObj } from '@storybook/react';
import Accordion from './Accordion';
import { ThemeContextProvider } from '../../theme/useTheme';

const meta: Meta<typeof Accordion> = {
  title: 'Components/Accordion',
  component: Accordion,
  decorators: [
    (Story: React.FC) => (
      <ThemeContextProvider>
        <Story />
      </ThemeContextProvider>
    ),
  ],
};

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    isOpen: false,
    disabled: false,
    size: 'medium',
  },
  render: (args) => (
    <Accordion {...args}>
      <Accordion.Heading>Heading</Accordion.Heading>
      <Accordion.Panel>Panel Content</Accordion.Panel>
    </Accordion>
  ),
};

export const OpenByDefault: Story = {
  args: {
    isOpen: true,
    disabled: false,
    size: 'medium',
  },
  render: (args) => (
    <Accordion {...args}>
      <Accordion.Heading>Heading</Accordion.Heading>
      <Accordion.Panel>Panel Content</Accordion.Panel>
    </Accordion>
  ),
};

export const Disabled: Story = {
  args: {
    isOpen: false,
    disabled: true,
    size: 'medium',
  },
  render: (args) => (
    <Accordion {...args}>
      <Accordion.Heading>Heading</Accordion.Heading>
      <Accordion.Panel>Panel Content</Accordion.Panel>
    </Accordion>
  ),
};

export const SmallSize: Story = {
  args: {
    isOpen: false,
    disabled: false,
    size: 'small',
  },
  render: (args) => (
    <Accordion {...args}>
      <Accordion.Heading>Heading</Accordion.Heading>
      <Accordion.Panel>Panel Content</Accordion.Panel>
    </Accordion>
  ),
};

export const CustomHeadingElement: Story = {
  args: {
    isOpen: false,
    disabled: false,
    size: 'medium',
  },
  render: (args) => (
    <Accordion {...args}>
      <Accordion.Heading as='h2'>
        Custom Heading as H2
      </Accordion.Heading>
      <Accordion.Panel>
        <p>This is the content of the accordion.</p>
      </Accordion.Panel>
    </Accordion>
  ),
};
