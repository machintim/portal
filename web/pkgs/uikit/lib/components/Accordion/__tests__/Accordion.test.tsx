import { describe, expect, test } from 'vitest';
import {
  render,
  screen,
  fireEvent,
} from '@testing-library/react';
import Accordion from '../Accordion';
import { ThemeContextProvider } from '../../../theme/useTheme';

describe('Accordion', () => {
  // Snapshot tests

  test('snapshot: no props', () => {
    const renderResult = render(
      <ThemeContextProvider>
        <Accordion>
          <Accordion.Heading>Heading</Accordion.Heading>
          <Accordion.Panel>Panel Content</Accordion.Panel>
        </Accordion>
      </ThemeContextProvider>
    );
    expect(
      renderResult.container.firstChild
    ).toMatchSnapshot();
  });

  test('snapshot: testId prop', () => {
    const renderResult = render(
      <ThemeContextProvider>
        <Accordion testId='abc'>
          <Accordion.Heading>Heading</Accordion.Heading>
          <Accordion.Panel>Panel Content</Accordion.Panel>
        </Accordion>
      </ThemeContextProvider>
    );
    expect(
      renderResult.container.firstChild
    ).toMatchSnapshot();
  });

  // Interaction tests

  test('should open and close the accordion', () => {
    render(
      <ThemeContextProvider>
        <Accordion>
          <Accordion.Heading>Heading</Accordion.Heading>
          <Accordion.Panel>Panel Content</Accordion.Panel>
        </Accordion>
      </ThemeContextProvider>
    );

    const heading = screen.getByText('Heading');
    expect(screen.queryByText('Panel Content')).toBeNull();

    fireEvent.click(heading);
    expect(
      screen.getByText('Panel Content')
    ).toBeInTheDocument();

    fireEvent.click(heading);
    expect(screen.queryByText('Panel Content')).toBeNull();
  });
});
