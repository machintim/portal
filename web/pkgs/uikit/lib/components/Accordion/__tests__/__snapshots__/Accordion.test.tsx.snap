// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion > snapshot: no props 1`] = `
<div
  aria-disabled="false"
  class="ucl-accordion css-1czwoas"
  data-testid="ucl-accordion"
>
  <div
    aria-expanded="false"
    class="ucl-accordion__heading css-11x7r0n"
    data-testid="ucl-accordion__heading"
    role="button"
    tabindex="0"
  >
    Heading
    <svg
      class="ucl-icon css-oi0ukk"
      data-testid="ucl-icon"
      fill="none"
      height="32"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="6 9 12 15 18 9"
      />
    </svg>
  </div>
</div>
`;

exports[`Accordion > snapshot: testId prop 1`] = `
<div
  aria-disabled="false"
  class="ucl-accordion css-1czwoas"
  data-testid="abc"
>
  <div
    aria-expanded="false"
    class="ucl-accordion__heading css-11x7r0n"
    data-testid="ucl-accordion__heading"
    role="button"
    tabindex="0"
  >
    Heading
    <svg
      class="ucl-icon css-oi0ukk"
      data-testid="ucl-icon"
      fill="none"
      height="32"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="6 9 12 15 18 9"
      />
    </svg>
  </div>
</div>
`;
